crates/duty-tracker/src/contract_state_machine.rs

```diff
@@ -213,6 +213,9 @@ pub enum ContractState {
     /// This state describes everything from the moment the deposit confirms, to the moment the
     /// strata state commitment that assigns this deposit confirms.
     Deposited {
+        /// The txid of the deposit request transaction that kicked off this contract.
+        deposit_request_txid: Txid,
```

Comment:
is there a reason you aren't using `ContractSM::deposit_request_txid`? If you go this route, please don't put it in state, put it in config where it belongs as this doesn't change for the lifetime of the contract.